### PR TITLE
test: Stop reloading sshd when Debian networking changes

### PR DESCRIPTION
**Pull request by stefwalter**
Whenever NetworkManager or anything changes the network, Debian and
Ubuntu reloads sshd with SIGHUP. This leads to a real connectivity
issue.

In our tests this leads to intermittent connection failures. In real
life this would result in an admin or Ansible unable to reach the
system temporarily.

We choose to work around this problem for now by disabling the
broken behavior. In reality the best fix for this problem would be
to have sshd use socket activation on Debian and Ubuntu, the way
it does on Fedora

---

Socket activation is not enabled in Fedora ([justification](https://bugzilla.redhat.com/show_bug.cgi?id=963268#c4)) and Debian ([justification](https://anonscm.debian.org/cgit/pkg-ssh/openssh.git/tree/debian/README.Debian)) by default.

Patch proposal to use  `IP_FREEBIND`: https://bugzilla.mindrot.org/show_bug.cgi?id=2512

Request to drop the hack from the Debian/Ubuntu packages: https://launchpad.net/bugs/1674330

 * [x] Ubuntu image
 * [ ] Debian 8 image
 * [x] Debian Testing

### CODE REVIEW

**Comment by stefwalter** [comment]
bot: Image refresh for debian-testing

**Comment by stefwalter** [comment]
bot: Image refresh for debian-8

**Comment by stefwalter** [comment]
bot: Image refresh for ubuntu-1604

**Comment by stefwalter** [comment]
One of the bugs this should fix:

```
# testBasic (check_loopback.TestLoopback)
#
Ignoring Resource Error: Operation canceled http://10.111.122.180:9090/cockpit/$2d0c79dd1407218715b1d8636a208af32f846470/base1/fonts/glyphicons.woff
Resource Error: Error downloading http://10.111.122.180:9090/cockpit/login - server replied: Authentication failed: no-cockpit http://10.111.122.180:9090/cockpit/login
> fatal: Authentication failed: no-cockpit
Warning: Permanently added '10.111.122.180' (ECDSA) to the list of known hosts.
Warning: Permanently added '10.111.122.180' (ECDSA) to the list of known hosts.
ssh.service is not active, cannot reload.
not ok 119 testBasic (check_loopback.TestLoopback) duration: 17s
Traceback (most recent call last):
  File "./verify/check-loopback", line 52, in testBasic
    m.execute("( ! systemctl is-active sshd.socket || systemctl stop sshd.socket) && systemctl restart sshd.service", direct=True)
  File "/build/cockpit/test/common/testvm.py", line 414, in execute
    raise subprocess.CalledProcessError(proc.returncode, command, output=output)
CalledProcessError: Command '( ! systemctl is-active sshd.socket || systemctl stop sshd.socket) && systemctl restart sshd.service' returned non-zero exit status 1
Wrote TestLoopback-testBasic-FAIL.png
Journal extracted to TestLoopback-testBasic-10.111.122.180-FAIL.log
```

Another one:

```
# testBasic (check_loopback.TestLoopback)
#
> Authentication failed: no-host
not ok 115 testBasic (check_loopback.TestLoopback) duration: 77s
Traceback (most recent call last):
  File "./verify/check-loopback", line 47, in testBasic
    b.wait_visible('#login-fatal')
  File "/build/cockpit/test/common/testlib.py", line 225, in wait_visible
    return self.wait_js_func('ph_is_visible', selector)
  File "/build/cockpit/test/common/testlib.py", line 210, in wait_js_func
    return self.phantom.wait("%s(%s)" % (func, ','.join(map(jsquote, args))))
  File "/build/cockpit/test/common/testlib.py", line 738, in <lambda>
    return lambda *args: self._invoke(name, *args)
  File "/build/cockpit/test/common/testlib.py", line 764, in _invoke
    raise Error(res['error'])
Error: timeout
Authentication failed: no-host
Wrote TestLoopback-testBasic-FAIL.png
Journal extracted to TestLoopback-testBasic-10.111.122.42-FAIL.log
```

**Comment by cockpituous** [comment]
Image creation for ubuntu-1604 in process on cockpit-9.
Log: http://fedorapeople.org/groups/cockpit/logs/refresh-ubuntu-1604-0df5f62a451034e12e19b7515e9a6a0966e42960/

**Comment by cockpituous** [comment]
Image creation for debian-8 in process on cockpit-11.
Log: http://fedorapeople.org/groups/cockpit/logs/refresh-debian-8-0df5f62a451034e12e19b7515e9a6a0966e42960/

**Comment by cockpituous** [comment]
Image creation for debian-testing in process on cockpit-7.
Log: http://fedorapeople.org/groups/cockpit/logs/refresh-debian-testing-0df5f62a451034e12e19b7515e9a6a0966e42960/

**Comment by cockpituous** [comment]
Image creation done: https://github.com/cockpituous/cockpit/commits/refresh-ubuntu-1604-0df5f62a451034e12e19b7515e9a6a0966e42960

**Comment by cockpituous** [comment]
Image creation failed.

**Comment by stefwalter** [comment]
bot: Image refresh for debian-8

**Comment by cockpituous** [comment]
Image creation for debian-8 in process on cockpit-9.
Log: http://fedorapeople.org/groups/cockpit/logs/refresh-debian-8-0df5f62a451034e12e19b7515e9a6a0966e42960/

**Comment by cockpituous** [comment]
Image creation stopped to avoid conflict.

**Comment by cockpituous** [comment]
Image creation done: https://github.com/cockpituous/cockpit/commits/refresh-debian-8-0df5f62a451034e12e19b7515e9a6a0966e42960

File: test/images/debian-testing [hunk 1]
```diff
@@ -1 +0,0 @@
-debian-testing-f272d4495e3d91e03c2cea8c4e302e7211fd8480.qcow2
```
Comment (martinpitt) [inline test/images/debian-testing]: This looks wrong -- why does the image link disappear completely? It should just be replaced with a new image sha?

**Comment by stefwalter** [comment]
bot: Image refresh for debian-testing

**Comment by cockpituous** [comment]
Image creation for debian-testing in process on cockpit-7.
Log: http://fedorapeople.org/groups/cockpit/logs/refresh-debian-testing-89af1a1a4d01032fbce7f4fdec68a2d788a70d20/

**Comment by cockpituous** [comment]
Image creation aborted

**Comment by martinpitt** [comment]
debian-testing image failed to build with `E: Unable to locate package libpcp3`. This is fallout from commit f7527bf465f3af4360a4da4b7fce5ee5c4c4efbd, this accidentally reintroduced the dependency into test/images/scripts/debian-testing.setup. I'll do a followup push to re-fix.

**Comment by martinpitt** [comment]
Please git am http://piware.de/tmp/0001-test-Re-drop-pcp-from-debian-testing.setup.patch here, I can't push to your repo.

**Comment by stefwalter** [comment]
Thanks @martinpitt

**Comment by stefwalter** [comment]
bot: Image refresh for debian-testing

**Comment by cockpituous** [comment]
Image creation for debian-testing in process on cockpit-11.
Log: http://fedorapeople.org/groups/cockpit/logs/refresh-debian-testing-6612ed2e339650a493582331515aa906cecfe212/

**Comment by cockpituous** [comment]
Image creation done.
Branch creation failed.

**Comment by martinpitt** [comment]
The "branch creation failed" might be due to the removed debian-testing symlink. However, the image was built and uploaded correctly, so I suggest to just manually update the link in your branch:

https://fedorapeople.org/groups/cockpit/images/debian-testing-090c9f9f798435dd071a83fa18310c797b356bce.qcow2.xz

File: test/images/debian-testing (hunk 1 above)
Comment (stefwalter) [inline test/images/debian-testing]: Reverted this change.

**Comment by stefwalter** [comment]
bot: Image refresh for debian-testing

**Comment by cockpituous** [comment]
Image creation for debian-testing in process on cockpit-7.
Log: http://fedorapeople.org/groups/cockpit/logs/refresh-debian-testing-3f44ee42e2bc4dc794c3679a1dff0b96a77ac871/

**Comment by cockpituous** [comment]
Image creation done: https://github.com/cockpituous/cockpit/commits/refresh-debian-testing-3f44ee42e2bc4dc794c3679a1dff0b96a77ac871

File: test/images/debian-testing (hunk 1 above)
Comment (stefwalter) [inline test/images/debian-testing]: Updated image.

**Comment by martinpitt** [comment]
(Other than the commit message which is still misleading, but not that important..)

**Comment by martinpitt** [comment]
My bad:
```
dpkg-checkbuilddeps: Unmet build dependencies: libpolkit-agent-1-dev
```
Commit f7527b removed it, but I didn't put it back in commit 1c579ffebc when restoring.

**Comment by martinpitt** [comment]
@stefwalter : Can you please git am http://piware.de/tmp/0001-test-Put-back-polkit-agent-build-dependency-into-Deb.patch on top of this and rebuild images again?

**Comment by stefwalter** [comment]
bot: Image refresh for debian-testing

**Comment by stefwalter** [comment]
bot: Image refresh for debian-8

**Comment by stefwalter** [comment]
bot: Image refresh for ubuntu-1604

**Comment by cockpituous** [comment]
Image creation for debian-testing in process on cockpit-9.
Log: http://fedorapeople.org/groups/cockpit/logs/refresh-debian-testing-cc0ddd0f0b7011a2e0ffa73c147b9c2be10f723e/

**Comment by cockpituous** [comment]
Image creation for debian-testing in process on cockpit-11.
Log: http://fedorapeople.org/groups/cockpit/logs/refresh-debian-testing-cc0ddd0f0b7011a2e0ffa73c147b9c2be10f723e/

**Comment by cockpituous** [comment]
Image creation aborted

**Comment by cockpituous** [comment]
Image creation for debian-testing in process on cockpit-9.
Log: http://fedorapeople.org/groups/cockpit/logs/refresh-debian-testing-cc0ddd0f0b7011a2e0ffa73c147b9c2be10f723e/

**Comment by cockpituous** [comment]
Image creation aborted

**Comment by cockpituous** [comment]
Image creation for debian-testing in process on cockpit-11.
Log: http://fedorapeople.org/groups/cockpit/logs/refresh-debian-testing-cc0ddd0f0b7011a2e0ffa73c147b9c2be10f723e/

**Comment by cockpituous** [comment]
Image creation aborted

**Comment by cockpituous** [comment]
Image creation for debian-testing in process on cockpit-9.
Log: http://fedorapeople.org/groups/cockpit/logs/refresh-debian-testing-cc0ddd0f0b7011a2e0ffa73c147b9c2be10f723e/

**Comment by cockpituous** [comment]
Image creation aborted

**Comment by cockpituous** [comment]
Image creation for debian-testing in process on cockpit-11.
Log: http://fedorapeople.org/groups/cockpit/logs/refresh-debian-testing-cc0ddd0f0b7011a2e0ffa73c147b9c2be10f723e/

**Comment by cockpituous** [comment]
Image creation aborted

**Comment by cockpituous** [comment]
Image creation for debian-testing in process on cockpit-9.
Log: http://fedorapeople.org/groups/cockpit/logs/refresh-debian-testing-cc0ddd0f0b7011a2e0ffa73c147b9c2be10f723e/

**Comment by cockpituous** [comment]
Image creation aborted

**Comment by cockpituous** [comment]
Image creation for debian-testing in process on cockpit-11.
Log: http://fedorapeople.org/groups/cockpit/logs/refresh-debian-testing-cc0ddd0f0b7011a2e0ffa73c147b9c2be10f723e/

**Comment by cockpituous** [comment]
Image creation aborted

**Comment by cockpituous** [comment]
Image creation for debian-testing in process on cockpit-9.
Log: http://fedorapeople.org/groups/cockpit/logs/refresh-debian-testing-cc0ddd0f0b7011a2e0ffa73c147b9c2be10f723e/

**Comment by cockpituous** [comment]
Image creation aborted

**Comment by cockpituous** [comment]
Image creation for debian-testing in process on cockpit-11.
Log: http://fedorapeople.org/groups/cockpit/logs/refresh-debian-testing-cc0ddd0f0b7011a2e0ffa73c147b9c2be10f723e/

**Comment by cockpituous** [comment]
Image creation for debian-testing in process on cockpit-7.
Log: http://fedorapeople.org/groups/cockpit/logs/refresh-debian-testing-cc0ddd0f0b7011a2e0ffa73c147b9c2be10f723e/

**Comment by cockpituous** [comment]
Image creation aborted

**Comment by cockpituous** [comment]
Image creation for debian-testing in process on cockpit-9.
Log: http://fedorapeople.org/groups/cockpit/logs/refresh-debian-testing-cc0ddd0f0b7011a2e0ffa73c147b9c2be10f723e/

**Comment by cockpituous** [comment]
Image creation aborted

**Comment by cockpituous** [comment]
Image creation for debian-testing in process on cockpit-11.
Log: http://fedorapeople.org/groups/cockpit/logs/refresh-debian-testing-cc0ddd0f0b7011a2e0ffa73c147b9c2be10f723e/

**Comment by cockpituous** [comment]
Image creation aborted

**Comment by cockpituous** [comment]
Image creation for debian-testing in process on cockpit-9.
Log: http://fedorapeople.org/groups/cockpit/logs/refresh-debian-testing-cc0ddd0f0b7011a2e0ffa73c147b9c2be10f723e/

**Comment by cockpituous** [comment]
Image creation for debian-testing in process on cockpit-10.
Log: http://fedorapeople.org/groups/cockpit/logs/refresh-debian-testing-cc0ddd0f0b7011a2e0ffa73c147b9c2be10f723e/

**Comment by cockpituous** [comment]
Image creation aborted

**Comment by cockpituous** [comment]
Image creation for debian-testing in process on cockpit-11.
Log: http://fedorapeople.org/groups/cockpit/logs/refresh-debian-testing-cc0ddd0f0b7011a2e0ffa73c147b9c2be10f723e/

**Comment by cockpituous** [comment]
Image creation aborted

**Comment by cockpituous** [comment]
Image creation for debian-testing in process on cockpit-9.
Log: http://fedorapeople.org/groups/cockpit/logs/refresh-debian-testing-cc0ddd0f0b7011a2e0ffa73c147b9c2be10f723e/

**Comment by cockpituous** [comment]
Image creation aborted

**Comment by cockpituous** [comment]
Image creation for debian-testing in process on cockpit-10.
Log: http://fedorapeople.org/groups/cockpit/logs/refresh-debian-testing-cc0ddd0f0b7011a2e0ffa73c147b9c2be10f723e/

**Comment by cockpituous** [comment]
Image creation aborted

**Comment by cockpituous** [comment]
Image creation for debian-testing in process on cockpit-11.
Log: http://fedorapeople.org/groups/cockpit/logs/refresh-debian-testing-cc0ddd0f0b7011a2e0ffa73c147b9c2be10f723e/

**Comment by cockpituous** [comment]
Image creation aborted

**Comment by cockpituous** [comment]
Image creation for debian-testing in process on cockpit-9.
Log: http://fedorapeople.org/groups/cockpit/logs/refresh-debian-testing-cc0ddd0f0b7011a2e0ffa73c147b9c2be10f723e/

**Comment by cockpituous** [comment]
Image creation aborted

**Comment by cockpituous** [comment]
Image creation for debian-testing in process on cockpit-10.
Log: http://fedorapeople.org/groups/cockpit/logs/refresh-debian-testing-cc0ddd0f0b7011a2e0ffa73c147b9c2be10f723e/

**Comment by cockpituous** [comment]
Image creation aborted

**Comment by cockpituous** [comment]
Image creation for debian-testing in process on cockpit-11.
Log: http://fedorapeople.org/groups/cockpit/logs/refresh-debian-testing-cc0ddd0f0b7011a2e0ffa73c147b9c2be10f723e/

**Comment by cockpituous** [comment]
Image creation done: https://github.com/cockpituous/cockpit/commits/refresh-debian-testing-cc0ddd0f0b7011a2e0ffa73c147b9c2be10f723e

**Comment by cockpituous** [comment]
Image creation for debian-testing in process on cockpit-7.
Log: http://fedorapeople.org/groups/cockpit/logs/refresh-debian-testing-cc0ddd0f0b7011a2e0ffa73c147b9c2be10f723e/

**Comment by cockpituous** [comment]
Image creation aborted

**Comment by cockpituous** [comment]
Image creation for debian-testing in process on cockpit-9.
Log: http://fedorapeople.org/groups/cockpit/logs/refresh-debian-testing-cc0ddd0f0b7011a2e0ffa73c147b9c2be10f723e/

**Comment by cockpituous** [comment]
Image creation for debian-testing in process on cockpit-8.
Log: http://fedorapeople.org/groups/cockpit/logs/refresh-debian-testing-cc0ddd0f0b7011a2e0ffa73c147b9c2be10f723e/

**Comment by cockpituous** [comment]
Image creation aborted

**Comment by cockpituous** [comment]
Image creation for debian-testing in process on cockpit-10.
Log: http://fedorapeople.org/groups/cockpit/logs/refresh-debian-testing-cc0ddd0f0b7011a2e0ffa73c147b9c2be10f723e/

**Comment by cockpituous** [comment]
Image creation aborted

**Comment by cockpituous** [comment]
Image creation for debian-testing in process on cockpit-11.
Log: http://fedorapeople.org/groups/cockpit/logs/refresh-debian-testing-cc0ddd0f0b7011a2e0ffa73c147b9c2be10f723e/

**Comment by cockpituous** [comment]
Image creation aborted

**Comment by cockpituous** [comment]
Image creation aborted

**Comment by cockpituous** [comment]
Image creation for debian-testing in process on cockpit-9.
Log: http://fedorapeople.org/groups/cockpit/logs/refresh-debian-testing-cc0ddd0f0b7011a2e0ffa73c147b9c2be10f723e/

**Comment by cockpituous** [comment]
Image creation for debian-testing in process on cockpit-8.
Log: http://fedorapeople.org/groups/cockpit/logs/refresh-debian-testing-cc0ddd0f0b7011a2e0ffa73c147b9c2be10f723e/

**Comment by cockpituous** [comment]
Image creation aborted

**Comment by cockpituous** [comment]
Image creation for debian-testing in process on cockpit-10.
Log: http://fedorapeople.org/groups/cockpit/logs/refresh-debian-testing-cc0ddd0f0b7011a2e0ffa73c147b9c2be10f723e/

**Comment by cockpituous** [comment]
Image creation aborted

**Comment by cockpituous** [comment]
Image creation for ubuntu-1604 in process on cockpit-11.
Log: http://fedorapeople.org/groups/cockpit/logs/refresh-ubuntu-1604-cc0ddd0f0b7011a2e0ffa73c147b9c2be10f723e/

**Comment by cockpituous** [comment]
Image creation aborted

**Comment by cockpituous** [comment]
Image creation for ubuntu-1604 in process on cockpit-8.
Log: http://fedorapeople.org/groups/cockpit/logs/refresh-ubuntu-1604-cc0ddd0f0b7011a2e0ffa73c147b9c2be10f723e/

**Comment by cockpituous** [comment]
Image creation aborted

**Comment by cockpituous** [comment]
Image creation for ubuntu-1604 in process on cockpit-9.
Log: http://fedorapeople.org/groups/cockpit/logs/refresh-ubuntu-1604-cc0ddd0f0b7011a2e0ffa73c147b9c2be10f723e/

**Comment by cockpituous** [comment]
Image creation aborted

**Comment by cockpituous** [comment]
Image creation aborted

**Comment by cockpituous** [comment]
Image creation for ubuntu-1604 in process on cockpit-11.
Log: http://fedorapeople.org/groups/cockpit/logs/refresh-ubuntu-1604-cc0ddd0f0b7011a2e0ffa73c147b9c2be10f723e/

**Comment by cockpituous** [comment]
Image creation for ubuntu-1604 in process on cockpit-10.
Log: http://fedorapeople.org/groups/cockpit/logs/refresh-ubuntu-1604-cc0ddd0f0b7011a2e0ffa73c147b9c2be10f723e/

**Comment by cockpituous** [comment]
Image creation aborted

**Comment by cockpituous** [comment]
Image creation for ubuntu-1604 in process on cockpit-8.
Log: http://fedorapeople.org/groups/cockpit/logs/refresh-ubuntu-1604-cc0ddd0f0b7011a2e0ffa73c147b9c2be10f723e/

**Comment by cockpituous** [comment]
Image creation aborted

**Comment by cockpituous** [comment]
Image creation aborted

**Comment by cockpituous** [comment]
Image creation for ubuntu-1604 in process on cockpit-11.
Log: http://fedorapeople.org/groups/cockpit/logs/refresh-ubuntu-1604-cc0ddd0f0b7011a2e0ffa73c147b9c2be10f723e/

**Comment by cockpituous** [comment]
Image creation for ubuntu-1604 in process on cockpit-10.
Log: http://fedorapeople.org/groups/cockpit/logs/refresh-ubuntu-1604-cc0ddd0f0b7011a2e0ffa73c147b9c2be10f723e/

**Comment by cockpituous** [comment]
Image creation stopped to avoid conflict.

**Comment by cockpituous** [comment]
Image creation for ubuntu-1604 in process on cockpit-7.
Log: http://fedorapeople.org/groups/cockpit/logs/refresh-ubuntu-1604-cc0ddd0f0b7011a2e0ffa73c147b9c2be10f723e/

**Comment by cockpituous** [comment]
Image creation aborted

**Comment by martinpitt** [comment]
Something is running nuts here, I remove the bot label for now until we figure out what's going on. https://fedorapeople.org/groups/cockpit/images/ already has several image rebuilds, but somehow the image build process gets interrupted and restarted in a seemingly eternal loop.

**Comment by cockpituous** [comment]
Image creation aborted

**Comment by cockpituous** [comment]
Image creation aborted

**Comment by cockpituous** [comment]
Image creation aborted

**Comment by cockpituous** [comment]
Image creation done: https://github.com/cockpituous/cockpit/commits/refresh-ubuntu-1604-cc0ddd0f0b7011a2e0ffa73c147b9c2be10f723e

**Comment by martinpitt** [comment]
bot: Image refresh for debian-8

**Comment by martinpitt** [comment]
I force-pushed the new hashes of the ubuntu-1604 and debian-testing images that got built yesterday; I requested a debian-8 rebuild as that never finished, I'll watch this to ensure the bot doesn't wreak havoc again.

**Comment by cockpituous** [comment]
Image creation for ubuntu-1604 in process on cockpit-7.
Log: http://fedorapeople.org/groups/cockpit/logs/refresh-ubuntu-1604-897bbf332dcee903923ec6d2ac0d42f58eaf9cfd/

**Comment by cockpituous** [comment]
Image creation for ubuntu-1604 in process on cockpit-10.
Log: http://fedorapeople.org/groups/cockpit/logs/refresh-ubuntu-1604-897bbf332dcee903923ec6d2ac0d42f58eaf9cfd/

**Comment by cockpituous** [comment]
Image creation for ubuntu-1604 in process on cockpit-11.
Log: http://fedorapeople.org/groups/cockpit/logs/refresh-ubuntu-1604-897bbf332dcee903923ec6d2ac0d42f58eaf9cfd/

**Comment by cockpituous** [comment]
Image creation aborted

**Comment by martinpitt** [comment]
Argh it's doing it again. I'll build the debian-8 image manually.

**Comment by cockpituous** [comment]
Image creation aborted

**Comment by cockpituous** [comment]
Image creation aborted

**Comment by martinpitt** [comment]
Re-pushed with updated debian-8 image and adjusted commit log. @stefwalter, as I meddled a lot with that PR now, can you give this a final review please?

Setting priority as this fixes test issues, blocks three image rebuilds, and should be good now, so let's get this out of the way ASAP.

**Pull request merged**